app/services/charge_models/factory.rb

```diff
@@ -24,12 +28,12 @@ def self.new_instance(chargeable:, aggregation_result:, properties:, period_rati
       end
     end
 
-    def self.charge_model_class(chargeable:)
+    def self.charge_model_class(chargeable:, has_aggregator: true)
```

Comment:
It looks good to me. Maybe could you just add a small comment explaining the why of this new argument